lib/ansible/modules/database/mysql/mysql_user.py

```diff
@@ -434,7 +434,7 @@ def pick(x):
             return x
 
     for grant in grants:
-        res = re.match("GRANT (.+) ON (.+) TO '.*'@'.*'( IDENTIFIED BY PASSWORD '.+')? ?(.*)", grant[0])
+        res = re.match("GRANT (.+) ON (.+) TO ['`].*['`]@['`].*['`]( IDENTIFIED BY PASSWORD ['`].+['`])? ?(.*)", grant[0])
```

Comment:
how about using the following :

```
GRANT (.+) ON (.+) TO (['`]).*\3@(['`]).*\4( IDENTIFIED BY PASSWORD (['`]).+\5)? ?(.*)
```

This doesn't allow for things such as `foo'` (mixing backticks and 's)